modules/ROOT/pages/client-id-based-policies.adoc

```diff
@@ -14,48 +14,53 @@ endif::[]
 
-Credentials data from client application credentials registered to protected APIs, are provided by Anypoint Platform. Once a policy requiring such information is applied (eg Client ID enforcement policy), the Runtime caches persistently the information locally updating it periodically.
+== How the Policy Works
 
-Besides its performance benefits, this action enables the policy to work offline in the event of a sudden loss of connectivity to the platform. The enforcement is applied regardless of the state of the management plane, with existing data. Once connectivity is re-established, the cache will be periodically updated again.
+Every client application that wants to consume an API protected by a Client ID enforcement policy needs to have previously requested access to the API. After an approved contract exists between the application and the API, every request needs to include the client credentials in compliance with how the policy has been configured.
 
-= REST Console
-To prevent user requests from being rejected when using the REST Console, create a trait in the RAML root and then reference it in every operation of your API affected by policy. You may find the RAML snippet link containing the RAML code you need to add to the RAML in the corresponding policy from the list of applied policies in the Policies tab of your API specification in API Manager.
+For example, if the policy has been configured to expect a client ID and client secret as headers, the application needs to make sure that those credentials are sent in the request using the corresponding headers.
 
 image::raml-snippet.png[]
 
-= Credentials source
+== Configure the Policy
+
+You can configure the Client ID Enforcement policy to obtain the client credentials in two different ways, HTTP basic authentication header and custom expression. 
+
+*HTTP Basic Authentication Header*
+
+You can configure the policy to require credentials as part of the Authorization header by selecting the HTTP Basic Authentication option. In this case, the application consuming the API must use the basic authentication to send the credentials in the requests.
 
-The policy can be configured to extract the client ID and client secret from the HTTP request in different ways using DataWeave 2.0 expressions.
+*Custom Expression*
 
-The following examples assume that the values for application credentials are '1234' for client ID and 'abcd' for client secret.
+This options allows you to provide an expression each for client ID and client secret, specifying where to extract the credentials from the request. In Mule 3 you can provide an MEL Expression and in Mule 4 you must provide  a DataWeave 2.0 expression. Use this option when you want to send the credentials in a custom header or somewhere else in the payload. You can also configure the client secret requirement as optional by leaving the field blank.
```

Comment:
I'm not an english native, but is "allows you to provide an expression each for client ID and client secret" correct?

Reply:
There is an extra space in "must provide a DataWeave 2.0 expression"

Reply:
Yes for the first sentence. It means you need to provide one expression for client ID and one for client secret. Fixed the space issue. Thanks!